src/options/imports/index_ops.js

```diff
@@ -0,0 +1,56 @@
+import index from 'src/search/search-index'
+
+const db = index.db._db
+
+async function downloadChunk(data, counter) {
+    chrome.downloads.download({
+        url: URL.createObjectURL(
+            new Blob([JSON.stringify(data)], { type: 'text/plain' }),
+        ),
+        filename: 'worldbrain_index_'.concat(counter, '.json'),
+    })
+}
+
+async function grabDBChunk(startAfter = '', limit = 1000) {
+    const data = []
+
+    return new Promise((resolve, reject) =>
+        db
+            .createReadStream({ gt: startAfter, limit })
+            .on('data', datum => data.push(datum))
+            .on('error', reject)
+            .on('end', () => resolve(data)),
+    )
+}
+
+/**
+ * @param {int} chunkSize Length of the chunk for piecewise export 
+ *
+ */
+export async function exportIndex(chunkSize = 500) {
+    let startAfter, chunk
+    let counter = 1
+
+    do {
+        chunk = await grabDBChunk(startAfter, chunkSize)
+
+        await downloadChunk(chunk, counter)
+        counter = counter + 1
+        // Set `startAfter` to last key, so that next iteration will be new data
+        startAfter = chunk[chunk.length - 1].key
+    } while (chunk.length === chunkSize) // If chunk `length` prop is < `chunkSize`, means we've run out of data
+}
+
+/**
+ * @param {JSON Array} JSON array of the index data to be inserted
+ */
+export async function importIndex(data) {
+    data.forEach(datum => {
```

Comment:
`Array.prototype.forEach` won't wait for any async logic inside an iteration before proceeding to the next one. So for `data.length` of N, there can be up to N `db.put`s put on the event loop. You can use a `for..of` loop to use `await`s inside the loop block and limit calls some way.

May also be better to do a single batched put, using the batch interface. Then you could just queue up the `.put` calls on the batch, using sync collection methods like `forEach`, and have a final `await` when you run the batched async stuff. [Example](https://github.com/WorldBrain/Memex/blob/master/src/search/search-index/add.js#L116).